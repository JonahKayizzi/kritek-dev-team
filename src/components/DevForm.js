import exp from 'constants';
import React from 'react';

const DevForm = (props) => {
  const { addDevProp } = props;
  const [devItem, setDevItem] = React.useState({
    devName: '',
    devLanguage: '',
    devYearsOfExperience: '',
    devSpecialty: '',
    devEmploymentType: '',
    devTitle: '',
    devLocation: '',
    devSalary: '',
    devDateOfBirth: '',
  });

  const handleChange = (e) => {
    setDevItem({
      ...devItem,
      [e.target.name]: e.target.value,
    });
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    addDevProp(devItem);
    setDevItem({
      devName: '',
      devSpecialty: '',
      devEmploymentType: '',
      devLocation: '',
      devSalary: '',
      devDateOfBirth: '',
    });
  };

  return (
    <form onSubmit={handleSubmit}>
      <p>Add New Dev</p>
      <input
        type="text"
        placeholder="Developer Name ..."
        value={devItem.devName}
        onChange={handleChange}
        name="devName"
      />
      <input
        type="text"
        placeholder="Speciality e.g Full Stack, Frontend ..."
        value={devItem.devSpecialty}
        onChange={handleChange}
        name="devSpecialty"
      />
      <input
        type="text"
        placeholder="Employment Type e.g Freelance, Full Time ..."
        value={devItem.devEmploymentType}
        onChange={handleChange}
        name="devEmploymentType"
      />
      <input
        type="text"
        placeholder="Location"
        value={devItem.devLocation}
        onChange={handleChange}
        name="devLocation"
      />
      <input
        type="text"
        placeholder="Salary"
        value={devItem.devSalary}
        onChange={handleChange}
        name="devSalary"
      />
      <input
        type="text"
        placeholder="Date of Birth"
        value={devItem.devDateOfBirth}
        onChange={handleChange}
        name="devDateOfBirth"
      />
    </form>
  );
};

export default DevForm;

DevForm.propTypes = {
  addDevProp: exp.func.isRequired,
};
