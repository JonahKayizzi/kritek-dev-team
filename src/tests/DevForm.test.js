import { render } from '@testing-library/react';
import DevForm from '../components/DevForm';

describe('DevForm tests', () => {
  test('Snapshot test. Should render DevForm correctly', () => {
    const tree = render(<DevForm />);
    expect(tree).toMatchSnapshot();
  });
});
