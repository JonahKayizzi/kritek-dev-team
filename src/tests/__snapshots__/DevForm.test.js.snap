// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DevForm tests Snapshot test. Should render DevForm correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="border-b mb-2 pb-8 flex flex-row flex-wrap gap-5"
      >
        <p
          class="text-gray-400 text-xl font-semibold tracking-wide text-left w-full"
        >
          ADD NEW DEV
        </p>
        <input
          class="w-1/4 border border-gray-300 rounded-sm py-3 py-2 px-4 text-gray-700 leading-tight focus:outline-none focus:bg-white focus:border-gray-500"
          name="devName"
          placeholder="Developer Name ..."
          type="text"
          value=""
        />
        <input
          class="w-1/3 border border-gray-300 rounded-sm py-3 py-2 px-4 text-gray-700 leading-tight focus:outline-none focus:bg-white focus:border-gray-500"
          name="devSpecialty"
          placeholder="Speciality e.g Full Stack, Frontend ..."
          type="text"
          value=""
        />
        <input
          class="w-1/4 border border-gray-300 rounded-sm py-3 py-2 px-4 text-gray-700 leading-tight focus:outline-none focus:bg-white focus:border-gray-500"
          name="devEmploymentType"
          placeholder="Employment Type e.g Freelance, Full Time ..."
          type="text"
          value=""
        />
        <input
          class="w-1/4 border border-gray-300 rounded-sm py-3 py-2 px-4 text-gray-700 leading-tight focus:outline-none focus:bg-white focus:border-gray-500"
          name="devLocation"
          placeholder="Location"
          type="text"
          value=""
        />
        <input
          class="w-1/4 border border-gray-300 rounded-sm py-3 py-2 px-4 text-gray-700 leading-tight focus:outline-none focus:bg-white focus:border-gray-500"
          name="devSalary"
          placeholder="Salary"
          type="text"
          value=""
        />
        <input
          class="w-1/4 border border-gray-300 rounded-sm py-3 py-2 px-4 text-gray-700 leading-tight focus:outline-none focus:bg-white focus:border-gray-500"
          name="devDateOfBirth"
          placeholder="Date of Birth"
          type="text"
          value=""
        />
        <button
          class="font-semibold bg-cyan-500 text-white rounded-sm py-2 px-3"
          type="submit"
        >
          ADD DEVELOPER
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="border-b mb-2 pb-8 flex flex-row flex-wrap gap-5"
    >
      <p
        class="text-gray-400 text-xl font-semibold tracking-wide text-left w-full"
      >
        ADD NEW DEV
      </p>
      <input
        class="w-1/4 border border-gray-300 rounded-sm py-3 py-2 px-4 text-gray-700 leading-tight focus:outline-none focus:bg-white focus:border-gray-500"
        name="devName"
        placeholder="Developer Name ..."
        type="text"
        value=""
      />
      <input
        class="w-1/3 border border-gray-300 rounded-sm py-3 py-2 px-4 text-gray-700 leading-tight focus:outline-none focus:bg-white focus:border-gray-500"
        name="devSpecialty"
        placeholder="Speciality e.g Full Stack, Frontend ..."
        type="text"
        value=""
      />
      <input
        class="w-1/4 border border-gray-300 rounded-sm py-3 py-2 px-4 text-gray-700 leading-tight focus:outline-none focus:bg-white focus:border-gray-500"
        name="devEmploymentType"
        placeholder="Employment Type e.g Freelance, Full Time ..."
        type="text"
        value=""
      />
      <input
        class="w-1/4 border border-gray-300 rounded-sm py-3 py-2 px-4 text-gray-700 leading-tight focus:outline-none focus:bg-white focus:border-gray-500"
        name="devLocation"
        placeholder="Location"
        type="text"
        value=""
      />
      <input
        class="w-1/4 border border-gray-300 rounded-sm py-3 py-2 px-4 text-gray-700 leading-tight focus:outline-none focus:bg-white focus:border-gray-500"
        name="devSalary"
        placeholder="Salary"
        type="text"
        value=""
      />
      <input
        class="w-1/4 border border-gray-300 rounded-sm py-3 py-2 px-4 text-gray-700 leading-tight focus:outline-none focus:bg-white focus:border-gray-500"
        name="devDateOfBirth"
        placeholder="Date of Birth"
        type="text"
        value=""
      />
      <button
        class="font-semibold bg-cyan-500 text-white rounded-sm py-2 px-3"
        type="submit"
      >
        ADD DEVELOPER
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
